src/d3a/models/myco_matcher/pay_as_bid.py

```diff
@@ -0,0 +1,37 @@
+from d3a.constants import FLOATING_POINT_TOLERANCE
+from d3a.models.market.market_structures import BidOfferMatch
+from d3a.models.myco_matcher.base_matcher import BaseMatcher
+
+
+class PayAsBidMatcher(BaseMatcher):
+
+    def calculate_match_recommendation(self, bids, offers, current_time=None):
+        """
+        Pay as bid first
+        There are 2 simplistic approaches to the problem
+        1. Match the cheapest offer with the most expensive bid. This will favor the sellers
+        2. Match the cheapest offer with the cheapest bid. This will favor the buyers,
+           since the most affordable offers will be allocated for the most aggressive buyers.
+        """
+
+        return self._perform_pay_as_bid_match(bids, offers)
+
+    def _perform_pay_as_bid_match(self, bids, offers):
+        # Sorted bids in descending order
+        sorted_bids = self.sort_by_energy_rate(bids, True)
```

Comment:
sort_by_energy_rate should not be called with self as it it a static method

Reply:
I do not agree on that. Static methods can be called using self , exactly like normal methods. It also makes the code much less verbose than using `PayAsBidMycoMatcherClassName.sort_by_energy_rate`